### DISCUSSION TITLE
createURL is not persist with browser query parameter

### DISCUSSION THREAD

**Discussion by Piusha** [algolia/instantsearch#4538] [General] [answered]
Here is the scenario, I have filter with brands. I created a custom filter widget with connectRefinementList. Now When I enable routes I can see createUrl() in historyRouter() function invoking and adding new filtered brand to it. but when I refresh the page and re-select any brand createURL() have only newly created filter values. it is seems to like browser URL parameters are not persist with the roteState after browser reload.

ultimately routeState should sync with browser history parameters.

**Comment by Haroenv** [comment, marked as answer]
Hi @Piusha, can you create what you have so far in a sandbox environment? We have a template available for that purpose [here](https://codesandbox.io/s/github/algolia/create-instantsearch-app/tree/templates/instantsearch.js). Thanks!

Custom widgets should take createURL in account and persist upon reload

**Reply by Piusha** [reply]
Hi Yes you are correct I manage to make it work. I haven't put  index name in in return rooutestate before return data from the routeToState 

i.e
routeToState(routeState) {
// .......
//...

return {'algolia_index': {....routes}}

}

BTW it was pretty hard to find clear documentation over the internet. 
Thnx

**Reply by Haroenv** [reply]
ah yes, that's a common pitfall. We have a FAQ item on that already here: https://www.algolia.com/doc/guides/building-search-ui/troubleshooting/faq/js/#why-is-my-uistate-ignored but I'm not sure what could be clearer right now. We probably need a warning if routeToState returns nothing, while routeState isn't nothing
